client/state/sites/hooks/with-sites-sorting.tsx

```diff
@@ -0,0 +1,16 @@
+import { createHigherOrderComponent } from '@wordpress/compose';
+import { ComponentType } from 'react';
+import { useSitesSorting } from './use-sites-sorting';
+
+export const withSitesSortingPreference = createHigherOrderComponent(
+	< OuterProps, >( Component: ComponentType< OuterProps > ) => {
+		const ComponentWithSitesSorting: ComponentType<
+			Omit< OuterProps, keyof ReturnType< typeof useSitesSorting > >
+		> = ( props ) => {
+			return <Component { ...( props as OuterProps ) } { ...useSitesSorting() } />;
+		};
+
+		return ComponentWithSitesSorting;
+	},
+	'withSitesSortingPreference'
+);
```

Comment:
Is there a better way of doing this? What I'm trying to do is to pass the `useSitesSorting` props, but not require them in the usage of the wrapped component.

Reply:
@Automattic/team-calypso Any pointers you might be able to offer ^ ?